priam/src/main/java/com/netflix/priam/aws/SDBInstanceData.java

```diff
@@ -45,17 +50,24 @@
         public final static String HOSTNAME = "hostname";
     }
 
-    public static final String DOMAIN = "InstanceIdentity";
-    public static final String ALL_QUERY = "select * from " + DOMAIN + " where " + Attributes.APP_ID + "='%s'";
-    public static final String INSTANCE_QUERY = "select * from " + DOMAIN + " where " + Attributes.APP_ID + "='%s' and " + Attributes.LOCATION + "='%s' and " + Attributes.ID + "='%d'";
+    public static final String DEFAULT_DOMAIN = "InstanceIdentity";
+    public static final String ALL_QUERY = "select * from `%s` where " + Attributes.APP_ID + "='%s'";
+    public static final String INSTANCE_QUERY = "select * from `%s` where " + Attributes.APP_ID + "='%s' and " + Attributes.LOCATION + "='%s' and " + Attributes.ID + "='%d'";
 
     private final ICredential provider;
     private final IConfiguration configuration;
-
+    private final String domain;
+    
     @Inject
     public SDBInstanceData(ICredential provider, IConfiguration configuration) {
         this.provider = provider;
         this.configuration = configuration;
+        String configuredDomain = System.getProperty("priam.sdb.instanceidentity.domain");
```

Comment:
Can we wrap this configuration into IConfiguration?

Reply:
Sure; will do asap